api/chat.go

```diff
@@ -45,61 +45,124 @@ func configGinChatRouter(router *gin.RouterGroup) {
 		if tumLiveContext.User == nil {
 			return
 		}
-		var chat ChatReq
-		if err := json.Unmarshal(msg, &chat); err != nil {
-			return
-		}
-		if !tumLiveContext.Course.ChatEnabled {
-			return
-		}
-		uname := tumLiveContext.User.Name
-		if chat.Anonymous && tumLiveContext.Course.AnonymousChatEnabled {
-			uname = "Anonymous"
-		}
-		replyTo := sql.NullInt64{}
-		if chat.ReplyTo == 0 {
-			replyTo.Int64 = 0
-			replyTo.Valid = false
-		} else {
-			replyTo.Int64 = chat.ReplyTo
-			replyTo.Valid = true
-		}
-		chatForDb := model.Chat{
-			UserID:   strconv.Itoa(int(tumLiveContext.User.ID)),
-			UserName: uname,
-			Message:  chat.Msg,
-			StreamID: tumLiveContext.Stream.ID,
-			Admin:    tumLiveContext.User.ID == tumLiveContext.Course.UserID,
-			ReplyTo:  replyTo,
-		}
-		err := dao.AddMessage(&chatForDb)
+		var req wsReq
+		err := json.Unmarshal(msg, &req)
 		if err != nil {
-			if errors.Is(err, model.ErrCooledDown) {
-				sendServerMessage("You are sending messages too fast. Please wait a bit.", TypeServerErr, s)
-			}
+			log.Warn(err)
 			return
 		}
-		if broadcast, err := json.Marshal(chatForDb); err == nil {
-			broadcastStream(tumLiveContext.Stream.ID, broadcast)
+		if req.Type == "message" {
```

Comment:
Maybe we can use a `switch` here and handle an unknown message type in the `default` case?

Reply:
:+1: I made this a switch statement and log the invalid request type as `default`